### PR TITLE
- use long long instead of int for free disk space with MiB base

### PR DESCRIPTION
**Pull request by aschnell**
Found while analysing bsc #991090.

### CODE REVIEW

**Comment by wfeldt** [comment]
lgtm

File: src/NCPkgPopupDiskspace.cc [hunk 1]
```diff
@@ -232,7 +232,7 @@ void NCPkgDiskspace::checkRemainingDiskSpace( const ZyppPartitionDu & partition
     if ( totalSize != 0 )
 	percent = ( 100 * usedSize ) / totalSize;
 
-    int	free	= ( totalSize - usedSize ) / FSize::MB;
+    long long free = ( totalSize - usedSize ) / FSize::MB;
```
Comment (lslezak) [inline src/NCPkgPopupDiskspace.cc]: Would it be possible to use `FSize` type directly? Both `totalSize` and `usedSize` have that type already.

File: package/libyui-ncurses-pkg.changes [hunk 2]
```diff
@@ -1,3 +1,9 @@
+-------------------------------------------------------------------
+Mon Jun 25 09:58:08 CEST 2018 - aschnell@suse.com
+
+- use long long instead of int for free disk space with MiB base
```
Comment (mvidner) [inline package/libyui-ncurses-pkg.changes]: Please include the bug reference.
The comment describes the implementation but does not tell the user the effect or the reason for the change. How about "Fixed an erroneous warning about not enough space on exabyte-sized disks" ?

File: src/NCPkgPopupDiskspace.cc (hunk 1 above)
Comment (lslezak) [reply src/NCPkgPopupDiskspace.cc]: Oh, it's compared to `MIN_FREE_MB_PROXIMITY` which is in MB, so it must be converted to MB first...

So it's OK.

Comment (aschnell) [reply src/NCPkgPopupDiskspace.cc]: Sure, but free has MiB as base while FSize always B. So simply using FSize would be confusing. Using FSize would also need all the checks later to be changed.

**Pull request merged**